src/utils_lua.c

```diff
@@ -274,38 +235,200 @@ int luaC_pushvalue(lua_State *L, value_t v, int ds_type) /* {{{ */
   return 0;
 } /* }}} int luaC_pushvalue */
 
-int luaC_pushvaluelist(lua_State *L, const data_set_t *ds,
-                       const value_list_t *vl) /* {{{ */
+int luaC_pushlabelset(lua_State *L, label_set_t label) /* {{{ */
 {
+  /*
+    label_set_t will be mapped to the following Lua table:
+
+    label = {
+      [1] => { name => value },
+      ...
+      [N] => { ... }
+    }
+   */
   lua_newtable(L);
+  if (label.num > 0) {
```

Comment:
You can remove this `if` – if `label.num` is zero the loop will simply be skipped.

Reply:
Thanks, fixed now.